### PR TITLE
Adding nil check for GetSASURLOptions.format()

### PR DESCRIPTION
**Pull request by souravgupta-msft**
1. Added nil check in `GetSASURLOptions.format()` method.
2. Fixed Account SAS creation.

- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [x] Updates to [CHANGELOG.md][] are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/storage/azblob/service/client_test.go [hunk 1]
```diff
@@ -693,20 +693,20 @@ func (s *ServiceUnrecordedTestsSuite) TestSASServiceClient() {
...
 	svcClient, err := service.NewClientWithNoCredential(sasUrl, nil)
 	_require.Nil(err)
 
-	// mismatched container name
-	_, err = svcClient.CreateContainer(context.Background(), containerName+"002", nil)
-	_require.Error(err)
-	testcommon.ValidateBlobErrorCode(_require, err, bloberror.AuthenticationFailed)
+	// create container using SAS
+	_, err = svcClient.CreateContainer(context.Background(), containerName, nil)
+	_require.Nil(err)
```
Comment (jhendrixMSFT) [inline sdk/storage/azblob/service/client_test.go:706]: It's preferable to use `_require.NoError(error)`.  You don't have to change it now, but we should clean this up at some point (it's used everywhere).

**Review by jhendrixMSFT — APPROVED** [review]

**Review by siminsavani-msft — APPROVED** [review]
LGTM!

**Comment by tasherif-msft** [comment]
@souravgupta-msft when we start the release process is there an easy way to target changes prior to this? since we want this change in post GA, should we have some sort of a dev branch instead of directly merging to main?

**Comment by jhendrixMSFT** [comment]
Why do we not want this change as part of GA?  Looks like a potential panic that we should fix.

**Comment by tasherif-msft** [comment]
> Why do we not want this change as part of GA? Looks like a potential panic that we should fix.

my bad I thought this was a PR for something completely different :sweat_smile: (thought we were adding a new api) but yes this should be in GA!

**Review by tasherif-msft — APPROVED** [review]

**Review by vibhansa-msft — APPROVED** [review]

**Comment by vibhansa-msft** [comment]
These changes shall be part of GA.

**Pull request merged**